app/models/spree/variant_decorator.rb

```diff
@@ -0,0 +1,15 @@
+Spree::Variant.class_eval do
+  
+  alias_method :original_price_in, :price_in
+
+  validates :sale_price, numericality: { less_than: :price }, allow_blank: true
+
+  def price_in(currency)
+    return original_price_in(currency) unless sale_price.present?
+    Spree::Price.new(variant_id: self.id, amount: self.sale_price, currency: currency)
```

Comment:
@rishabhguptarishi what if sale price is not present for that variant?

Reply:
@rajneeshsharma9 IF SALE_PRICE IS NOT PRESENT THAN ORIGINAL PRICE_IN METHOD GETS CALLED

Reply:
@rishabhguptarishi use the new method you create for variant decoration in place of 

> amount: self.sale_price

Reply:
@rishabhguptarishi have you done above change?